docker-compose.yml

```diff
@@ -16,9 +15,50 @@ services:
       - MONGO_USER_READ_PASSWORD=${MONGO_USER_READ_PASSWORD}
       - MONGO_USER_ADMIN_USERNAME=${MONGO_USER_ADMIN_USERNAME}
       - MONGO_USER_ADMIN_PASSWORD=${MONGO_USER_ADMIN_PASSWORD}
-    ports:
-      - "27017:27017"
+    # ports:
+    #   - "27017:27017"
+    expose:
+      - ${MONGODB_PORT}
     volumes:
       - ./db/data:/data/db
       - ./db/scripts/mongo-init.js:/docker-entrypoint-initdb.d/mongo-init.js
     env_file: .env
+    networks:
+      - dev
+
+  frontend:
+    container_name: gallery_frontend
+    image: node:20-alpine
+    restart: unless-stopped
+    volumes:
+      - ./frontend:/frontend
+    expose:
+      - ${FRONTEND_PORT}
+    working_dir: /frontend  # Set the working directory to the project root
+    command: npm start -- --port=${FRONTEND_PORT}
+    env_file: .env
+    networks:
+      - dev
```

Comment:
[Question] Regarding to the hot reloading, try this (adding nodemodules to the volume)

```
services:
  frontend:
    container_name: gallery_frontend
    image: node:20-alpine
    restart: unless-stopped
    volumes:
      - ./frontend:/frontend
      - /frontend/node_modules # add node modules
    ports:
      - ${FRONTEND_PORT}:${FRONTEND_PORT} # port mapping
    working_dir: /frontend
    command: npm start -- --port=${FRONTEND_PORT}
    env_file: .env
    networks:
      - dev

  # other services...

networks:
  dev:
    driver: bridge
```